force-app/main/default/classes/ClassesObjectsMethods.cls

```diff
@@ -163,9 +168,12 @@ public with sharing class ClassesObjectsMethods {
      */
     public static Integer generateRandomNumber() {
         // Use Math class to generate number between 50 and 100
-
+        Integer randomNumber=0;
+        while(randomNumber<50){
+            randomNumber=Integer.valueof(Math.random()*100);
+        }
```

Comment:
This is sort of a brute force way to to do this. What about a random number 0-50 and then add 50 to it?